### ISSUE TITLE
Type 'ListProps' is not assignable to type 'ListControllerProps'

### ISSUE THREAD

**Issue by olegcherr** [marmelab/react-admin#7110]
**React-Admin v4.0.0-alpha.1**

TypeScript complains about wrong prop type for List/Edit/Show components:
```
Type '(props: ListProps<Record>) => Element' is not assignable to type 'ComponentType<ListControllerProps<Record>> | undefined'
Type '(props: ShowProps<Record>) => Element' is not assignable to type 'ComponentType<ShowControllerProps<Record>> | undefined'.
```
```
<Resource name="positions" list={PositionList} show={PositionShow} />

function PositionList(props: ListProps) {
...

function PositionShow(props: ShowProps) {
...
```

![image](https://user-images.githubusercontent.com/3730074/150192858-47df3cc9-73f7-41ae-a566-e568eafc94e0.png)

Related, I guess: https://github.com/marmelab/react-admin/issues/7109

UPD:
https://codesandbox.io/s/demo--ra-list-show-props-type-bug-0o4jt?file=/src/App.tsx

**Comment by WiXSL** [comment]
I cannot reproduce this error in simple example in next branch

**Comment by olegcherr** [comment]
I reproduced it on codepen:
https://codesandbox.io/s/demo--ra-list-show-props-type-bug-0o4jt?file=/src/App.tsx

![image](https://user-images.githubusercontent.com/3730074/150510238-ae0594b4-23f1-4771-8093-08ec47b39f5c.png)

**Comment by WiXSL** [comment]
I don't know why it doesn't show in the simple example, but its seems you are right.

**Comment by fzaninotto** [comment]
Fixed by #7121
